### PR TITLE
Switch GIT repo shouldn't error out if GIT repo exists

### LINKED ISSUE #1402: [git] Error logged when repository switcher in statusbar is clicked
> Uncaught (in promise) The command 'git.change.repository' cannot be executed. There are no active handlers available for the command.

Looks like the command handler is disabled if only one repository is available. Maybe in that case the handler should be active, but just do nothing.

### LINKED ISSUE #1455: [git] Regression in the Git status bar contribution
AC:
 - [ ] The currently active repository should not be visible when only one repository is available in the WS.
 - [ ] It should not throw an error when clicking on it. Currently: 
    ```
    The command 'git.change.repository' cannot be executed. There are no active handlers available for the command.
    CommandRegistry.executeCommand @ command.ts:169
    attrs.onclick @ status-bar.ts:157
    ```

### PR DESCRIPTION
**Pull request by elaihau**
- Fixes #1402
- Fixes #1455 

Signed-off-by: elaihau <liang.huang@ericsson.com>

### CODE REVIEW

**Comment by kittaakos** [comment]
We did not show the repository if there was only one.

**Comment by AlexTugarev** [comment]
@kittaakos you are right! shame on me, I screwed it up here: https://github.com/theia-ide/theia/commit/e6dd6f1421983e24a2406eb98d9358188d2772f1#diff-d86a105da42850a54d9cc2b0ad3af23aL52

@elaihau, if you're addressing this, please re-add the `repositories.length > 1` check.

**Comment by elaihau** [comment]
>  if you're addressing this, please re-add the repositories.length > 1 check.

Sure @AlexTugarev . i will do it. Thank you for pointing me to the file where the change is needed :) !

**Comment by AlexTugarev** [comment]
@elaihau that file was moved. new location is https://github.com/theia-ide/theia/blob/master/packages/git/src/browser/git-view-contribution.ts#L84, where you could query the tracker for `allRepositories`.

**Comment by elaihau** [comment]
I am thinking of a slightly more complicated scenario:

Step 1: Open the work space ~/dev, where there is only one GIT repo available: `~/dev/theia`.
Step 2: Create a second git repo under `~/dev`, say, running `~/dev> ng new helloWorld` from command line prompt.
Step 3: Watch what happens to the opened workspace in theia: 
   3.1) the new directory (i.e., helloworld) shows up in the file browser, however
   3.2) with the fix to #1455, the "current active GIT repo" button does not show up, or, without the fix to #1455, the new GIT repo (i.e., helloworld) does not show up in the quick open menu.

**Comment by AlexTugarev** [comment]
I think you're right, there is a gap. Even the refresh action in the Git view will not necessarily trigger such an update. 
I see, you already files #1549 👍

**Comment by elaihau** [comment]
This pull request is for #1455 and #1402. I will try to resolve #1549 in another PR.

File: packages/git/src/browser/git-view-contribution.ts [hunk 1]
```diff
@@ -215,4 +217,7 @@ export class GitViewContribution extends AbstractViewContribution<GitWidget> {
         return undefined;
     }
 
+    private hasMultipleRepositories(): boolean {
```
Comment (kittaakos) [inline packages/git/src/browser/git-view-contribution.ts]: Please go with the `protected` visibility.

File: packages/git/src/browser/git-view-contribution.ts [hunk 2]
```diff
@@ -215,4 +217,7 @@ export class GitViewContribution extends AbstractViewContribution<GitWidget> {
...
+        return this.repositoryTracker && this.repositoryTracker.allRepositories.length > 1;
```
Comment (kittaakos) [inline packages/git/src/browser/git-view-contribution.ts]: Why do you need to check the `repositoryTracker`? How could that be `undefined`?

**Comment by kittaakos** [comment]
I am reviewing right now.

File: packages/git/src/browser/git-view-contribution.ts [hunk 3]
```diff
@@ -160,8 +160,10 @@ export class GitViewContribution extends AbstractViewContribution<GitWidget> {
             isEnabled: () => !!this.repositoryTracker.selectedRepository
         });
         registry.registerCommand(GIT_COMMANDS.CHANGE_REPOSITORY, {
-            execute: () => this.quickOpenService.changeRepository(),
-            isEnabled: () => this.repositoryTracker.allRepositories.length > 1
+            execute: () => {
+                this.quickOpenService.changeRepository();
+            },
```
Comment (kittaakos) [inline packages/git/src/browser/git-view-contribution.ts]: FYI: you could write `execute: () => this.quickOpenService.changeRepository(),` too.

Comment (elaihau) [reply packages/git/src/browser/git-view-contribution.ts]: Sure will do

**Pull request merged**